libraries/libraries.go

```diff
@@ -0,0 +1,208 @@
+package libraries
+
+import (
+	"github.com/src-d/go-borges"
+	"github.com/src-d/go-borges/util"
+
+	"gopkg.in/src-d/go-errors.v1"
+)
+
+var (
+	// ErrLibraryExists an error returned when a borges.Library
+	// added before is attempted to be added again.
+	ErrLibraryExists = errors.NewKind("library %s already exists")
+)
+
+// FilterLibraryFunc stands for a borges.Library filter function.
+type FilterLibraryFunc func(borges.Library) (bool, error)
+
+// Libraries is an implementation to aggregate borges.Library in just one instance.
+// The borges.Library that will be added shouldn't contain other libraries inside.
+type Libraries struct {
+	libs map[borges.LibraryID]borges.Library
+}
+
+var _ borges.Library = (*Libraries)(nil)
+
+// New create a new Libraries instance.
+func New() *Libraries {
+	return &Libraries{
+		libs: map[borges.LibraryID]borges.Library{},
+	}
+}
+
+// Add adds a new borges.Library. It will fail with ErrLibraryExists
+// if the library was already added.
+func (l *Libraries) Add(lib borges.Library) error {
+	_, ok := l.libs[lib.ID()]
+	if ok {
+		return ErrLibraryExists.New(lib.ID())
+	}
+
+	l.libs[lib.ID()] = lib
+	return nil
+}
+
+// ID implements the Library interface.
+func (l *Libraries) ID() borges.LibraryID {
+	return ""
+}
+
+// Init implements the Library interface.
+func (l *Libraries) Init(borges.RepositoryID) (borges.Repository, error) {
+	return nil, borges.ErrNotImplemented.New()
+}
+
+// Get implements the Library interface.
+func (l *Libraries) Get(id borges.RepositoryID, mode borges.Mode) (borges.Repository, error) {
+	var repos []borges.Repository
+	for _, lib := range l.libs {
+		r, err := lib.Get(id, mode)
+		if err != nil {
+			if borges.ErrRepositoryNotExists.Is(err) {
+				continue
+			}
+
+			return nil, err
+		}
+
+		repos = append(repos, r)
+	}
+
+	if len(repos) == 0 {
+		return nil, borges.ErrRepositoryNotExists.New(id)
+	}
+
+	return repos[0], nil
+}
+
+// GetOrInit implements the Library interface.
+func (l *Libraries) GetOrInit(borges.RepositoryID) (borges.Repository, error) {
+	return nil, borges.ErrNotImplemented.New()
+}
+
+// Has implements the Library interface.
+func (l *Libraries) Has(id borges.RepositoryID) (bool, borges.LibraryID, borges.LocationID, error) {
+	var matches []*hasReturnValues
+	for _, lib := range l.libs {
+		has, libID, locID, err := lib.Has(id)
+		if err != nil {
+			return false, "", "", err
+		}
+
+		if has {
+			matches = append(matches, &hasReturnValues{
```

Comment:
Same as with `Get`. Should return as soon as one is found.

Reply:
Same as the previous comment